### ISSUE TITLE
Calculating target wights using neural network

### ISSUE THREAD

**Issue by yamada-yuka0322** [yamada-yuka0322/PFS_imaging#11]
![Image](https://github.com/user-attachments/assets/5cf65334-aa92-401c-a109-31294a30ef51)

I've been trying different learning rates, different number of layers and nodes, and different optimizers and is getting similar results as the figure above. I was expecting the dots to lie on top of the red line which is basically true density = predicted density.
But I'm starting to think that the result I got is actually reasonable because,

1. The distribution of the training data is concentrated near the mean density with smaller number of heal pixels with extremely high or low target density. This causes the prediction to bias towards the mean density, over estimating the target density for low density healpixels and under estimating the target density for high density pixels.

2. The dispersion of the true value should include the imaging systematics + the cosmological fluctuation while the prediction only include the imaging systematics, so the dispersion of the prediction should be smaller than the true data. So, maybe it is reasonable to have a distribution that looks elongated in the x axis.

**Comment by yamada-yuka0322** [comment]
Ver1.
PCA up to 0.99 (11 components)
model:11→1024→512→256→64→1
patience: 20
lr: 1e-5

![Image](https://github.com/user-attachments/assets/0830df80-0921-4747-84e2-8b66007c28c5)
![Image](https://github.com/user-attachments/assets/225f013e-a305-42f4-91e1-76f0e94b6ec3)
![Image](https://github.com/user-attachments/assets/d0c357e8-5e1f-489f-a463-86e2309723be)

**Comment by yamada-yuka0322** [comment]
Ver2.
PCA up to 0.9 (6 components)
model:11→1024→512→256→64→1
patience: 20
lr: 1e-5

![Image](https://github.com/user-attachments/assets/53e2c3ed-1145-429a-8e76-473717a57d02)
![Image](https://github.com/user-attachments/assets/a290a4df-ab89-4606-bbe1-bed4e4877063)
![Image](https://github.com/user-attachments/assets/3112a7c1-c71c-4107-8f79-c501e5de6129)

**Comment by yamada-yuka0322** [comment]
Ver0
No PCA
model:11→1024→512→256→64→1
patience: 20
lr: 1e-5

![Image](https://github.com/user-attachments/assets/39018cb2-48be-4d39-8c0a-157bf4de5eec)
![Image](https://github.com/user-attachments/assets/a509aa0e-2076-417a-ad0c-d7e693175696)
![Image](https://github.com/user-attachments/assets/9aac4d98-5ba2-4851-8110-f7b6b3888c8c)

**Comment by changhoonhahn** [comment]
@yamada-yuka0322 Lets use github issues as a way of efficiently communicating the results (i.e. takeaways), rather than listing the full process or dumping figures.  

First, you have to include legends in the scatter plots. I'd also recommend titles and text to make it clearer. Any unclear figure only causes more confusion. 

Also, include a brief but complete description of each test, the motivation for them, and the takeaway.

**Comment by yamada-yuka0322** [comment]
![Image](https://github.com/user-attachments/assets/2bd1a721-119d-4103-a5a3-958f88d9c0a4)

![Image](https://github.com/user-attachments/assets/c4c3e1d8-116e-4107-9846-65175c49a529)

I plotted the density fluctuation and angular power spectrum after hyper parameter tuning with optima. I saved the best model in a file and used the exact weight and biases.

**Comment by yamada-yuka0322** [comment]
<img width="2987" height="1041" alt="Image" src="https://github.com/user-attachments/assets/ac5ae76d-e1dd-4684-beac-ae8eff05b7c6" />
I calculated the angular power spectrum using weights and bias from the linear regression (yellow) and the power spectrum after reweighting using the linear regression weights.

I think it looks good because the black line and the yellow line agrees well, and the power spectrum with the neural network weights(blue) looks similar with the reweighted power spectrum.

<img width="2987" height="1041" alt="Image" src="https://github.com/user-attachments/assets/ae548462-fc49-4360-9c60-a3a83b64390e" />

**Comment by yamada-yuka0322** [comment]
<img width="2480" height="936" alt="Image" src="https://github.com/user-attachments/assets/7aaa4ad9-7bfc-456f-8ab3-09c1863ddbfe" />
I also calculated the correlation function with the same setup. The purple line goes below the black and yellow line, which is good, but it's hard for me to judge if the yellow line matches the black line. 

<img width="2458" height="936" alt="Image" src="https://github.com/user-attachments/assets/2c95c672-578f-4489-bfda-7cb975fc37c0" />
I think we do see a decrease in the correlation after applying the weights, for most of the imaging systematics,.
